src/hpp/corbaserver/rbprm/__main__.py

```diff
@@ -0,0 +1,74 @@
+import subprocess
+import time
+import atexit
+import argparse
+from importlib import import_module
+import sys
+import os
+
+
+def kill_process(proc):
+    proc.kill()
+
+
+# init argument parser
+parser = argparse.ArgumentParser(description="This script must be called with one argumemnt : "
+                                 "the name of the script describing the scenario."
+                                 "If a relative path is given, "
+                                 "hpp.corbaserver.rbprm.scenarios is prepended")
+parser.add_argument('scenario_name',
+                    type=str,
+                    help="The name of the scenario script to run. "
+                    "If a relative path is given, hpp.corbaserver.rbprm.scenarios is prepended")
+args = parser.parse_args()
+# retrieve argument
+scenario_name = args.scenario_name
+scenario_name = scenario_name.rstrip(".py")  # remove extension if specified
+scenario_name.replace("/", ".")  # if the path to the file is given, transform it to a python path
+# try to import the given module
+try:
+    module_scenario = import_module(scenario_name)
+except ImportError:
+    print("Cannot import " + scenario_name + " try to prepend path")
+scenario_name = "hpp.corbaserver.rbprm.scenarios." + scenario_name
+try:
+    module_scenario = import_module(scenario_name)
+except ImportError:
+    print("Cannot import " + scenario_name + ". Check if the path is correct")
+    sys.exit(1)
+
+# kill already existing instance of the viewer/server
+subprocess.run(["killall", "gepetto-gui"])
+subprocess.run(["killall", "hpp-rbprm-server"])
+# run the viewer/server in background
+# stdout and sterr outputs of the child process are redirected to denull (hidden).
+# preexec_fn is used to ignore ctrl-c signal send to the main script (otherwise they are forwarded to the child process)
+process_viewer = subprocess.Popen("gepetto-gui",
+                                  stdout=subprocess.PIPE,
+                                  stderr=subprocess.DEVNULL,
+                                  preexec_fn=os.setpgrp)
+process_server = subprocess.Popen("hpp-rbprm-server",
+                                  stdout=subprocess.PIPE,
+                                  stderr=subprocess.DEVNULL,
+                                  preexec_fn=os.setpgrp)
+# wait a little for the initialization of the server/viewer
+time.sleep(3)
+
+# Get ContactGenerator or PathPlanner class from the imported module and run it
```

Comment:
I'm not sure we need this block, or that we should require users to name their contact generator `ContactGenerator` while `TalosFlatGroundContactGenerator` could be a better choice.

2 other options come to my mind right now:

1. Change the `if __name__ == "__main__":` to `if __name__ == "__main__" or int(os.environ.get('RBPRM_RUN', 0))` and use a `os.environ['RBPRM_RUN'] = 1` before the `import_module(scenario_name)`
2. launch `getattr(module_scenario, 'main')()`, and in the scenario:
```python
class TalosFlatGroundContactGenerator(AbstractContactGenerator):
    ...

def main():
    contact_generator = TalosFlatGroundContactGenerator()
    contact_generator.run()

if __name__ == '__main__':
    main()
```

Reply:
I am not sure about this changes. It will work in this package, but there are other package that will import scenarios script from rbprm (SL1M or MLP for example).
In this packages I need to get the Class defined in the scripts to instantiate it and do other stuff than just calling `run()`.

I am not sure how I can handle that without having always the same class name in the scenarios scripts.

Reply:
I don't totally get your point, but it's totally OK to merge this PR without resolving this right now :)